### PR TITLE
Add videopressGUID attribute and use it.

### LINKED ISSUE #3803: Store video press guid on dedicated attribute
At the moment we are storing the video press ID on the shortcode attribute on media. 

This is not correct because the shortcode attribute should be used to store the html code to represent the media.

We should create a dedicate attribute for video id and use it to store the video press id.

### PR DESCRIPTION
**Pull request by SergioEstevao**
Closes #3803 

Needs Review: @bummytime

### CODE REVIEW

**Comment by bummytime** [comment]
Looks good @SergioEstevao. I tested the lightweight migration to v31 in various ways...seemed to work like a champ. I don't think we have a hard and fast rule, but maybe consider a migration unit test ala `CoreDataMigrationTests`? :-) 

Other than that everything works fine. :shipit:

**Comment by bummytime** [comment]
@SergioEstevao Actually, I hate to do this, but I am taking back that :shipit: . I am not sure if we need to create a new version of the data model (e.g. v32) or not. Perhaps @sendhil or @astralbodies could provide insight here....

File: WordPress/Classes/Services/MediaService.m [hunk 1]
```diff
@@ -213,16 +213,22 @@ - (void)getMediaURLFromVideoPressID:(NSString *)videoPressID
                             success:(void (^)(NSString *videoURL, NSString *posterURL))success
                             failure:(void (^)(NSError *error))failure
 {
-    NSSet *mediaSet = [blog.media filteredSetUsingPredicate:[NSPredicate predicateWithFormat:@"shortcode = %@", videoPressID]];
+    NSSet *mediaSet = [blog.media filteredSetUsingPredicate:[NSPredicate predicateWithFormat:@"videopressGUID = %@", videoPressID]];
```
Comment (astralbodies) [inline WordPress/Classes/Services/MediaService.m]: So I know this is how it was done before but I'd rather we query Core Data for the media items with a `NSFetchRequest`. I could see this getting strange if we got to the point of syncing all media (or maybe we do) and this causes a lot of faults to fill since its post-Core Data.

File: WordPress/Classes/Services/MediaService.m [hunk 2]
```diff
@@ -213,16 +213,22 @@ - (void)getMediaURLFromVideoPressID:(NSString *)videoPressID
...
     Media *media = [mediaSet anyObject];
     if (media) {
         if([[NSFileManager defaultManager] fileExistsAtPath:media.thumbnailLocalURL isDirectory:nil]) {
-            success(media.remoteURL, media.thumbnailLocalURL);
+            if (success){
```
Comment (astralbodies) [inline WordPress/Classes/Services/MediaService.m]: It's a nitpick but there should be a space between the `)` and `{`

File: WordPress/Classes/Services/MediaService.m [hunk 3]
```diff
@@ -213,16 +213,22 @@ - (void)getMediaURLFromVideoPressID:(NSString *)videoPressID
...
+                success(media.remoteURL, media.thumbnailLocalURL);
+            }
         } else {
-            success(media.remoteURL, @"");
+            if (success){
```
Comment (astralbodies) [inline WordPress/Classes/Services/MediaService.m]: Space between the `)` and `{`

File: WordPress/Classes/Services/MediaService.m [hunk 4]
```diff
@@ -213,16 +213,22 @@ - (void)getMediaURLFromVideoPressID:(NSString *)videoPressID
...
+                success(media.remoteURL, @"");
+            }
         }
     } else {
-        failure(nil);
+        if (failure){
```
Comment (astralbodies) [inline WordPress/Classes/Services/MediaService.m]: Space between the `)` and `{`

File: WordPress/Classes/Services/MediaService.m [hunk 5]
```diff
@@ -325,6 +331,7 @@ - (void)updateMedia:(Media *)media withRemoteMedia:(RemoteMedia *)remoteMedia
     media.width = remoteMedia.width;
     //media.exif = remoteMedia.exif;
```
Comment (astralbodies) [inline WordPress/Classes/Services/MediaService.m]: Commented code - delete?

File: WordPress/Classes/ViewRelated/Post/WPPostViewController.m [hunk 6]
```diff
@@ -1692,7 +1692,7 @@ - (void)uploadMedia:(Media *)media trackingId:(NSString *)mediaUniqueId
             [self.editorView replaceLocalVideoWithID:mediaUniqueId
                                       forRemoteVideo:media.remoteURL
                                         remotePoster:media.thumbnailLocalURL
-                                          videoPress:media.shortcode];
+                                          videoPress:media.videopressGUID];
```
Comment (astralbodies) [inline WordPress/Classes/ViewRelated/Post/WPPostViewController.m]: You might not have control over this but the parameter `videoPress` does not provide enough context. The other parameters are questionable. I know its outside the scope of this PR but it would provide clarity if it were something like:

```objc
- (void)replaceLocalVideoWithID:(NSString *)uniqueID
              forRemoteVideoURL:(NSString *)videoURL
                remotePosterURL:(NSString *)posterURL
                   videoPressID:(NSString *)videoPressID;
```

Something for @bummytime maybe? Not sure who "owns" that.

File: WordPress/Classes/WordPress.xcdatamodeld/WordPress 31.xcdatamodel/contents [hunk 7]
```diff
@@ -248,6 +248,7 @@
         <attribute name="title" optional="YES" attributeType="String">
             <userInfo/>
         </attribute>
+        <attribute name="videopressGUID" optional="YES" attributeType="String" syncable="YES"/>
```
Comment (astralbodies) [inline WordPress/Classes/WordPress.xcdatamodeld/WordPress 31.xcdatamodel/contents]: Modifying `WordPress 31.xcdatamodel` should be fine as we haven't done an internal build yet.

**Comment by bummytime** [comment]
Thanks for taking a peek at this @astralbodies !

File: WordPress/Classes/WordPress.xcdatamodeld/WordPress 31.xcdatamodel/contents (hunk 7 above)
Comment (SergioEstevao) [inline WordPress/Classes/WordPress.xcdatamodeld/WordPress 31.xcdatamodel/contents]: @astralbodies @bummytime this will actually need to be changed because meanwhile @aerych updated the model in the 5.3 release branch. So I will need to merge the release 5.3 and then create a new data model version (32) and update this branch.

Comment (astralbodies) [inline WordPress/Classes/WordPress.xcdatamodeld/WordPress 31.xcdatamodel/contents]: Yeah, I figured we'd have to resolve that after 5.3 is released. We could cherry pick the commits or bring develop current. @sendhil if you want me to work on merging the branch let me know. :smile:

**Comment by astralbodies** [comment]
@SergioEstevao Once @aerych is done merging release/5.3 into develop, maybe merge develop into this PR and I'll give you a :+1: with that merge.

**Comment by SergioEstevao** [comment]
@astralbodies ready for another look.

File: WordPress/Classes/ViewRelated/Post/WPPostViewController.m (hunk 6 above)
Comment (SergioEstevao) [inline WordPress/Classes/ViewRelated/Post/WPPostViewController.m]: This will need a change in the editor pod. Will be done on another PR on the editor.

**Comment by astralbodies** [comment]
:boom: :shipit:
